### PR TITLE
Update kube options

### PR DESCRIPTION
**Pull request by pavelpereyma**
Support MKS Cluster feature gates & admission controllers #166 .
Added data sources that could be using to get available feature gates or admission controllers for the specific Kubernetes versions:

- selectel_mks_available_feature_gates_v1
- selectel_mks_available_admission_controllers_v1

Updated dependencies:
- github.com/hashicorp/terraform-plugin-sdk/v2 -> v2.7.0
- github.com/selectel/mks-go -> v0.8.0

### CODE REVIEW

File: selectel/data_source_selectel_mks_kube_options.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kube_version_minor": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"names": {
+							Type:     schema.TypeString,
```
Comment (kolsean) [inline selectel/data_source_selectel_mks_kube_options.go:55]: It is not useful to fetch all feature gate names as one joined string.
Let's change it to `schema.TypeSet` like you did it for "selectel_mks_cluster_v1" resource.

**Review by kolsean — CHANGES_REQUESTED** [review]
Not bad! But we need to make some changes.

**Comment by pavelpereyma** [comment]
Thank you for your review. I made necessary changes,  please review changes and back if need something to fix.

File: selectel/data_source_selectel_mks_kube_options.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			"feature_gates": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kube_version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"names": {
```
Comment (kolsean) [inline selectel/data_source_selectel_mks_kube_options.go:54]: The `Elem` field must be set, otherwise it cause an internal error when trying to run the provider plugin.

The same error occurs for admission controllers.

Please look at you implementation at selectel/resource_selectel_mks_cluster_v1.go#134

**Review by kolsean — CHANGES_REQUESTED** [review]
When trying to create new data source with feature gates an internal error occured:
`Error: Invalid address to set: []string{"feature_gates", "0", "kube_version_minor"}`
Please check.

Also there is no documentation for new data sources.
Please add the doc.

File: selectel/resource_selectel_mks_cluster_v1_test.go [hunk 3]
```diff
@@ -59,6 +62,8 @@ func TestAccMKSClusterV1Basic(t *testing.T) {
 					resource.TestCheckResourceAttr("selectel_mks_cluster_v1.cluster_tf_acc_test_1", "enable_pod_security_policy", "false"),
 					resource.TestCheckResourceAttr("selectel_mks_cluster_v1.cluster_tf_acc_test_1", "maintenance_window_start", maintenanceWindowStartUpdated),
 					resource.TestCheckResourceAttr("selectel_mks_cluster_v1.cluster_tf_acc_test_1", "status", "ACTIVE"),
+					resource.TestCheckResourceAttr("selectel_mks_cluster_v1.cluster_tf_acc_test_1", "feature_gates.0", "TTLAfterFinished"),
```
Comment (kolsean) [inline selectel/resource_selectel_mks_cluster_v1_test.go:65]: It also would be great to pass AC & FG at `create` test step and  add the similar check for it.

File: website/docs/d/mks_feature_gates_v1.html.markdown [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+```hcl
+resource "selectel_vpc_project_v2" "project_1" {
+  auto_quotas = true
+}
+
+data "selectel_mks_feature_gates_v1" "fg" {
+  project_id = "${selectel_vpc_project_v2.project_1.id}"
+  region = "ru-3"
+  filter {
+    kube_version = "1.22."
```
Comment (kolsean) [inline website/docs/d/mks_feature_gates_v1.html.markdown:24]: Patch version is missed. Please add it or keep just a minor version without dot in the end.

File: website/docs/r/mks_cluster_v1.html.markdown [hunk 5]
```diff
@@ -72,6 +72,10 @@ The following arguments are supported:
     Argument "enable_patch_version_auto_upgrade" should be explicitly set to false in case of zonal cluster.
     Changing this creates a new cluster.
 
+* `feature_gates` - (Optional) Represents a set of feature gate names to be enabled in a Kubernetes cluster.
+
+* `admission_controllers` - (Optional) Represents a set of feature gate names to be enabled in a Kubernetes cluster.
```
Comment (kolsean) [inline website/docs/r/mks_cluster_v1.html.markdown:77]: a set of admission controllers

File: website/docs/d/mks_feature_gates_v1.html.markdown [hunk 6]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (kolsean) [inline website/docs/d/mks_feature_gates_v1.html.markdown:1]: Need to add the link on thin document to `website/selectel.erb` file.

**Review by kolsean — CHANGES_REQUESTED** [review]

File: website/docs/d/mks_admission_controllers_v1.html.markdown [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `admission_controllers` - Contains a list of the found available admission controllers.
+
+**admission_controllers**
+
+- `kube_version_minor` - Kubernetes version.
```
Comment (kolsean) [inline website/docs/d/mks_admission_controllers_v1.html.markdown:51]: Need to rename to `kube_version`

File: website/docs/d/mks_feature_gates_v1.html.markdown [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `feature_gates` - Contains a list of the found available feature gates.
+
+**feature_gates**
+
+- `kube_version_minor` - Kubernetes version.
```
Comment (kolsean) [inline website/docs/d/mks_feature_gates_v1.html.markdown:51]: Need to rename to `kube_version`

**Review by kolsean — CHANGES_REQUESTED** [review]
There are a few linter and doc issues left to fix.
`dupl threshold` in the linters settings can be raised to `300`

File: selectel/resource_selectel_mks_cluster_v1_test.go [hunk 9]
```diff
@@ -271,3 +282,167 @@ func testAccMKSClusterV1Zonal(projectName, clusterName, kubeVersion, maintenance
...
+
+func newTestMKSClient(n string, s *terraform.State) (*v1.ServiceClient, error) {
+	rs, ok := s.RootModule().Resources[n]
+	if !ok {
+		return nil, fmt.Errorf("not found: %s", n)
+	}
+
+	if rs.Primary.ID == "" {
+		return nil, errors.New("no ID is set")
+	}
```
Comment (kolsean) [inline selectel/resource_selectel_mks_cluster_v1_test.go:404]: These checks logically are not related to the creation of the MKS client, it's a little confusing. You should not include them in this function.

File: selectel/resource_selectel_mks_cluster_v1_test.go [hunk 10]
```diff
@@ -29,13 +31,20 @@ func TestAccMKSClusterV1Basic(t *testing.T) {
...
+	admissionControllersUpdate := defaultAdmissionContollers[1:2]
+
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { testAccSelectelPreCheck(t) },
 		ProviderFactories: testAccProviders,
 		CheckDestroy:      testAccCheckVPCV2ProjectDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccMKSClusterV1Basic(projectName, clusterName, kubeVersion, maintenanceWindowStart),
+				Config: testAccMKSClusterV1BasicWithKubeOptions(projectName, clusterName, kubeVersion, maintenanceWindowStart, featureGates, admissionControllers),
```
Comment (kolsean) [inline selectel/resource_selectel_mks_cluster_v1_test.go:47]: Function renaming is not needed, because we do not test different basic cases.

**Review by kolsean — CHANGES_REQUESTED** [review]

**Review by kolsean — APPROVED** [review]
LGTM!
